starlite/config/logging.py

```diff
@@ -49,17 +52,55 @@ class LoggingConfig(BaseModel):
     """This will be the configuration for the root logger. Processing of the configuration will be as for any logger,
     except that the propagate setting will not be applicable."""
 
-    def configure(self) -> None:
+    def _enable_rich_logging(self, log_config_dict: Dict[str, Any]) -> Dict[str, Any]:
+        """Overwrite the default console log configuration with rich log
+        configuration.
+
+        Args:
+            log_config_dict (Dict[str,Any]): dict representation of LoggingConfig model
+        Returns:
+            Dict[str,Any]: dict representation of LoggingConfig model with rich console logging in place of default console logging
+        """
+        log_config_dict["formatters"]["rich"] = {"format": "%(name)s - %(message)s"}
+
+        log_config_dict["handlers"].pop("console", None)  # remove console handler if present
+
+        log_config_dict["handlers"]["p_rich"] = {
+            "class": "rich.logging.RichHandler",
+            "level": "DEBUG",
+            "rich_tracebacks": True,
+            "formatter": "rich",
+        }
+        if "cfg://handlers.console" in log_config_dict["handlers"]["queue_listener"]["handlers"]:
+            log_config_dict["handlers"]["queue_listener"]["handlers"].remove("cfg://handlers.console")
+
+        log_config_dict["handlers"]["queue_listener"]["handlers"].append("cfg://handlers.p_rich")
+
+        return log_config_dict
+
+    def configure(self, state: "State") -> None:
         """Configured logger with the given configuration.
 
         If the logger class contains the word `picologging`, we try to
         import and set the dictConfig
         """
+        log_config_dict = self.dict(exclude_none=True)
+
+        if getattr(state, "rich_logging", False):
```

Comment:
Perphaps its simply possible to enhance the pydantic model itself with some values - such as a boolean flag for enabling rich logging. Then there will not be a need for saving values in state.

What you would then do is exclude these properties when calling self.dict.

Reply:
new commit addresses this